meson.build

```diff
@@ -1902,29 +1902,27 @@ basic_includes = include_directories(
         'src/systemd',
         '.')
 
-libsystemd_includes = [basic_includes, include_directories(
-        'src/libsystemd/sd-bus',
-        'src/libsystemd/sd-device',
-        'src/libsystemd/sd-event',
-        'src/libsystemd/sd-hwdb',
-        'src/libsystemd/sd-id128',
-        'src/libsystemd/sd-journal',
-        'src/libsystemd/sd-netlink',
-        'src/libsystemd/sd-network',
-        'src/libsystemd/sd-resolve')]
+libsystemd_includes = [
+        basic_includes,
+        include_directories(
+                'src/libsystemd/internal',
```

Comment:
Hm, I cannot understand why moving type 3 functions to libsystemd/internal can save file size...
And, I am not a fun of the idea. libsystemd/internal now has variety of codes...
Moreover, e.g. sd_device_monitor and sd_device_enumerator are exported but moved to internal. That's not intuitive and confusing for me...

Reply:
> Hm, I cannot understand why moving type 3 functions to libsystemd/internal can save file size... 

No, the moving of functions isn't supposed to save file size. Size is saved because we stop installing certain compiled objects twice. Namely, the public (`_public_`) functions that form libsystemd. The moving of (other) functions to src/libsystemd/internal is done so we can link them into *both* libsystemd and non-libsystemd-parts without pulling in libsystemd into the non-systemd parts. (Or in other words, src/libsystemd/internal now is used like src/libsystemd was used as a whole previously.)

I tried to express this in the commit message. I think the text *does* explain this, but I'm open to suggestions.

> Moreover, e.g. sd_device_monitor and sd_device_enumerator are exported but moved to internal.

That was an error in the patch. `device-monitor.c` and `device-enumerator.c` are now back under `src/libsystemd/sd-device/` and some of the code that was there is in `src/libsystemd/internal/device-internal.c`.

> And, I am not a fun of the idea. libsystemd/internal now has variety of codes...  That's not intuitive and confusing for me...

This is generally how the rest of shared code is organized. `src/fundamental`, `src/basic`, and `src/shared` all consist of code that has a variety of purposes, grouped by what's it's for, not what it does. I'm not trying to say that the split between files, or the file naming, or ordering of things within files is ideal. I mostly tried to minimize changes, e.g. keeping the same order when moving text between files, but I'm sure this could be improved. But this kind of cleanup (especially reordering within files or renaming of files) can be easily done later.